include/blogPost.php

```diff
@@ -1,62 +1,52 @@
-          ),
-           array(
-               'postId'=>2,
-               'title'=> "What I'm Listening to <br/>",
-               'body'=> "<div class = 'aboutMe'>
-                    Explore the album covers to see what I am listening to!<br/><br/><br/>",
-              ));
-          return $post;
-}
-// include_once('config/init.php');
-// function getAllBlogPostsPost($postID){
-// $Result = dbQuery("
-//     SELECT *
-//     FROM blog_post
-//     WHERE blogPostId =:blogPostId
-//     ", array("blogPostId"=>$postId));
-//     return $Result;
-// }
-
-function echoPostLinkHtml(){
-    $post = getAllBlogPosts();
-    for ($i = 0; $i<count($post); $i++){
-        $postId = $post[$i]['postId'];
-        $postTitle = $post[$i]['title'];
-        $postBody= $post[$i]['body'];
-      echo  "<a class = 'resume' href = 'viewPost.php?postId=". $postId."'>".$postTitle." </a><br/>";
+include_once('config/init.php');
+function blogTagOrganizer (){
+    $tagTypes = dbQuery("
+            SELECT *
+            FROM tags
+            ")->fetchALL();
+    $tagArray = array();
+        foreach ($tagTypes as $tagsOrganized){
+            if(!in_array($tagsOrganized['tagName'], $tagArray)){
+                array_push($tagArray, $tagsOrganized['tagName']);
+                echo "<a class = 'resume' href = 'listOfBlogs.php?tagPostId=". $tagsOrganized['tagId']."'>".$tagsOrganized['tagName']." </a>";
+                }
+            }
+        }
+function echoPostLinkHtml($postTagId){
+    $tagIdOrganizer = dbQuery("
+        SELECT * FROM blog_post
+        INNER JOIN blogPost_tags_linked ON blogPost_tags_linked.blogPostId = blog_post.blogPostId
+        INNER JOIN tags ON tags.tagId =  blogPost_tags_linked.tagId
+        WHERE tags.tagId = :varTagName
+        ", array("varTagName" => $postTagId))->fetchALL();
+        $tagPostArray = array();
+        foreach ($tagIdOrganizer as $tagPosts){
+            if(!in_array($tagPosts['tagName'], $tagPostArray)){
+                echo "<h4 class='projects'>".$tagPosts['tagName']."</h4>";
+                array_push($tagPostArray, $tagPosts['tagName']);
+            }
+            echo  "<a class = 'resume' href = 'viewPost.php?blogPostId=". $tagPosts['blogPostId']."'>".$tagPosts['title']." </a>";
+        }
     }
-}
-
-
-
 function contentOfBlog($postId){
-    $post = getAllBlogPosts();
-    $postTitle = $post[$postId]['title'];
-    $postBody= $post[$postId]['body'];
-    return "<h2>". $postTitle. "</h2>" . $postBody. "<br> <p style= 'text-decoration: underline'> Feel free to leave any comments below </p>
-     Name: <input type='text' name='name' id='name' /><br />
-    Email: <input type='text' name='email' id='email' /><br />
-     Comment:<br />
-     <textarea name='comment' id='comment'></textarea><br />
-     <input type='hidden' name='articleid' id='articleid' value=".$postId." />
-     <input type='submit' value='Submit' /> </form> </div>";
+    $post = dbQuery("
```

Comment:
This is fine for now, but I bet you'll want to separate this into its own function (something like `getBlogPost`) so that you can use it elsewhere. Right now you'd need to re-write this db query if you want to get a blog post anywhere else, and you generally want to avoid re-writing identical queries.